plansys2_planner/src/plansys2_planner/PlannerNode.cpp

```diff
@@ -156,4 +162,18 @@ PlannerNode::get_plan_service_callback(
   }
 }
 
+void
+PlannerNode::validate_domain_service_callback(
+  const std::shared_ptr<rmw_request_id_t> request_header,
+  const std::shared_ptr<plansys2_msgs::srv::ValidateDomain::Request> request,
+  const std::shared_ptr<plansys2_msgs::srv::ValidateDomain::Response> response)
+{
+  response->success = solvers_.begin()->second->isDomainValid(
+    request->domain, get_namespace());
+
+  if (!response->success) {
+    response->error_info = "Domain is not valid";
+  }
```

Comment:
Another option is for `isValidDomain()` to return a string error message that can also be tacked on to the service response...